### ISSUE TITLE
hrtime return type issue on PHP 8.0 if phpVersion is set to < 7.3

### ISSUE THREAD

**Issue by kelunik** [vimeo/psalm#5033]
https://github.com/amphp/amp/runs/1714053018?check_suite_focus=true#step:9:14

When run on PHP 8.0, an error is reported, because Psalm thinks the return type of `hrtime(false)` is `int` instead of `int[]`. But this only happens if the `phpVersion` in `psalm.xml` is set below `7.3`. Works fine on PHP 7.4.

Unrelated to the hrtime issue, I had to somehow add the function files to `<stubs />` when upgrading to Psalm 4.x for some reason.

```xml
<stubs>
    <file name="lib/functions.php" />
    <file name="lib/Internal/functions.php" />
</stubs>
```

**Comment by orklah** [comment]
@weirdan Can you take at look at this one?

https://psalm.dev/r/fb5c5555dc
https://psalm.dev/r/fb5c5555dc?php=7.2

I think this could be fixed by conditional execution again. The error reported by psalm is not really relevant because the code only use the function when it's available

**Comment by weirdan** [comment]
> I think this could be fixed by conditional execution again.

Should be easy, just make the body of this `if` unconditional:

https://github.com/vimeo/psalm/blob/ef0d19ed9fc66152e48229d4a0d305f19d1a6aa2/src/Psalm/Internal/PhpVisitor/Reflector/ExpressionResolver.php#L243
